packages/sdk/shell/src/panels/IdentityPanel/useAgentHandlers.ts

```diff
@@ -11,54 +11,91 @@ import { useAgentHostingClient, useClient } from '@dxos/react-client';
 import { type Identity } from '@dxos/react-client/halo';
 import { Invitation, InvitationEncoder } from '@dxos/react-client/invitations';
 
+import { type AgentFormProps } from '../../components';
+
 export const useAgentHandlers = ({
   invitations,
   identity,
 }: {
   invitations: CancellableInvitation[];
   identity: Identity | null;
-}) => {
+}): AgentFormProps => {
   const [validationMessage, setValidationMessage] = useState('');
-  const [agentStatus, setAgentStatus] = useState('');
-  const [agentActive, setAgentActive] = useState(false);
+  const [agentStatus, setAgentStatus] = useState<AgentFormProps['agentStatus']>('getting');
   const agentHostingProviderClient = useAgentHostingClient();
   const client = useClient();
 
-  useEffect(() => {
-    const fetchAgentStatus = async () => {
-      // TODO(nf): if agent exists, subscribe to updates instead of oneshot status
-      if (agentHostingProviderClient) {
-        const agentStatus = await agentHostingProviderClient?.getAgent(identity?.identityKey.truncate() ?? 'foo');
-        if (agentStatus) {
-          setAgentActive(true);
-          setAgentStatus(agentStatus);
+  const updateAgentStatus = async () => {
+    // TODO(nf): if agent exists, subscribe to updates instead of oneshot status
+    if (agentHostingProviderClient) {
+      setAgentStatus('getting');
+      try {
+        const agentId = await agentHostingProviderClient?.getAgent(identity?.identityKey.truncate() ?? 'never');
+        if (agentId) {
+          setAgentStatus('online');
```

Comment:
I think part of the idea of integrating the agent with the device list and having it show up as a special device is that it should use similar signals as well. I'm not sure what the difference between online/ready is, but I wouldn't expect it to be "online" unless I can see it's presence as active in my devices.

Reply:
I’ve adjusted how this works — the managed agent will appear in the usual list if it is in `devices`, otherwise `AgentConfig` will render.